### ISSUE TITLE
Error while using make

### ISSUE THREAD

**Issue by duck7000** [maxrd2/subtitlecomposer#88]
Compiling from source gives this error :
```
In member function ‘SubtitleComposer::SString& SubtitleComposer::SString::replace(const QRegularExpression&, const SubtitleComposer::SString&, bool)’:
/home/ed/Downloads/subtitlecomposer-0.6.5/src/core/sstring.cpp:1031:61: error: ‘qAsConst’ was not declared in this scope
   for(const SStringCapture &backRef: qAsConst(backReferences)) {
```

I'm using Linux mint 18.3 and tried to follow your wiki.

Maybe it's possible to provide deb files for ubuntu, Linuxmint? This would make things a lot easier for a starting Linux user like myself

Edit: I'm using the latest master (10-27-2017)

edit2: The error is also in the latest release (0.6.5)

**Comment by maxrd2** [comment]
qAsConst() function was introduced in Qt 5.7.
Which Qt version are you using?

**Comment by duck7000** [comment]
As far as i can tell qt 5.5 so that is probably the reason for the error i guess
i think that updating qt is not possible, i'm using a LTS version (mint 18 is based on ubuntu 16.04 LTS)

I will try your suggested link

i can't help with the deb scripts, lack of knowledge so i hope that someone else has that knowledge

**Comment by muzena** [comment]
@duck7000 
Try this ppa: https://launchpad.net/~trebelnik-stefina/+archive/ubuntu/subtitlecomposer
sudo add-apt-repository ppa:trebelnik-stefina/subtitlecomposer
sudo apt-get update
sudo apt-get install subtitlecomposer

I add in preinstall script ppa for Qt5.7 dependancies (Qt 5.9)
https://launchpad.net/~trebelnik-stefina/+archive/ubuntu/radeon-profile-dependencies

**Comment by maxrd2** [comment]
@duck7000 
if missing qAsConst is only error you get during compile, you can edit source and change qAsConst(...) to (...), it will work the same.

**Comment by duck7000** [comment]
@muzena 
I installed your verison 0.6.5 , it installed perfectly but when i start subtitle composer it still says 0.6.4?

I don't know how to use your second suggestion about the preinstall script

@maxrd2 
I will edit the source and try again

**Comment by duck7000** [comment]
After editing the source this error is gone but raises a few others:
```
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp: In member function ‘bool SubtitleComposer::VobSubInputProcessDialog::Frame::processPieces()’:
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:2: error: ‘function’ is not a member of ‘std’
  std::function<void(int,int)> cutPiece = [&](int x, int y){
  ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:28: error: expression list treated as compound expression in functional cast [-fpermissive]
  std::function<void(int,int)> cutPiece = [&](int x, int y){
                            ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:16: error: expected primary-expression before ‘void’
  std::function<void(int,int)> cutPiece = [&](int x, int y){
                ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:176:18: error: ‘cutPiece’ was not declared in this scope
     cutPiece(x, y);
                  ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:138:6: warning: unused variable ‘bgColor’ [-Wunused-variable]
  int bgColor = pieceBitmap.pixel(0, 0);

```
I used the 0.6.5 release version

**Comment by muzena** [comment]
@duck7000 
I compile wrong source 0.6.3

**Comment by muzena** [comment]
@maxrd2 
What is qAsConst?
Can be added somewhere or is it needed, option like this to be compiled with Qt 5.9, I had similar problem with compiling Radeon Profile it depends on min Qt 5.6.
https://github.com/marazmista/radeon-profile/pull/65/files
I added Qt 5.9 PPA and these options like in radeon Profile README file, and then compilation with Qt 5.9 worked.

I have same compilation errors like @duck7000

**Comment by maxrd2** [comment]
@muzena It just casts variable type to const type, so it can be omitted. It might produce a bit slower code but won't change much what code does.
http://doc.qt.io/qt-5/qtglobal.html#qAsConst

**Comment by maxrd2** [comment]
@duck7000 Are you using (outdated) compiler that doesn't support c++11. What's your gcc version?

**Comment by duck7000** [comment]
gcc version is 5.4.0

**Comment by duck7000** [comment]
I managed to install gcc6 and c++-6 so the version is now 6.3.0 but still errors:
```
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp: In member function ‘bool SubtitleComposer::VobSubInputProcessDialog::Frame::processPieces()’:
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:2: error: ‘function’ is not a member of ‘std’
  std::function<void(int,int)> cutPiece = [&](int x, int y){
  ^~~
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:28: error: expression list treated as compound expression in functional cast [-fpermissive]
  std::function<void(int,int)> cutPiece = [&](int x, int y){
                            ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:16: error: expected primary-expression before ‘void’
  std::function<void(int,int)> cutPiece = [&](int x, int y){
                ^~~~
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:176:18: error: ‘cutPiece’ was not declared in this scope
     cutPiece(x, y);
                  ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:138:6: warning: unused variable ‘bgColor’ [-Wunused-variable]
  int bgColor = pieceBitmap.pixel(0, 0);
      ^~~~~~~

```

**Comment by duck7000** [comment]
If it helps here is the complete configure and make text

```
./configure
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at /usr/share/ECM/kde-modules/KDEInstallDirs.cmake:311 (message):
  LIB_INSTALL_DIR is deprecated, use KDE_INSTALL_LIBDIR instead.
Call Stack (most recent call first):
  /usr/share/ECM/kde-modules/KDEInstallDirs.cmake:405 (_define_relative)
  CMakeLists.txt:17 (include)


-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Found KF5Auth: /usr/lib/x86_64-linux-gnu/cmake/KF5Auth/KF5AuthConfig.cmake (found version "5.18.0") 
-- Found KF5Config: /usr/lib/x86_64-linux-gnu/cmake/KF5Config/KF5ConfigConfig.cmake (found version "5.18.0") 
-- Found KF5ConfigWidgets: /usr/lib/x86_64-linux-gnu/cmake/KF5ConfigWidgets/KF5ConfigWidgetsConfig.cmake (found version "5.18.0") 
-- Found KF5CoreAddons: /usr/lib/x86_64-linux-gnu/cmake/KF5CoreAddons/KF5CoreAddonsConfig.cmake (found version "5.18.0") 
-- Found Gettext: /usr/bin/msgmerge (found version "0.19.7") 
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Found KF5I18n: /usr/lib/x86_64-linux-gnu/cmake/KF5I18n/KF5I18nConfig.cmake (found version "5.18.0") 
-- Found KF5KIO: /usr/lib/x86_64-linux-gnu/cmake/KF5KIO/KF5KIOConfig.cmake (found version "5.18.0") 
-- Found KF5XmlGui: /usr/lib/x86_64-linux-gnu/cmake/KF5XmlGui/KF5XmlGuiConfig.cmake (found version "5.18.0") 
-- Found KF5Sonnet: /usr/lib/x86_64-linux-gnu/cmake/KF5Sonnet/KF5SonnetConfig.cmake (found version "5.18.0") 
-- Found KF5Kross: /usr/lib/x86_64-linux-gnu/cmake/KF5Kross/KF5KrossConfig.cmake (found version "5.18.0") 
-- Found KF5Codecs: /usr/lib/x86_64-linux-gnu/cmake/KF5Codecs/KF5CodecsConfig.cmake (found version "5.18.0") 
-- Found KF5TextWidgets: /usr/lib/x86_64-linux-gnu/cmake/KF5TextWidgets/KF5TextWidgetsConfig.cmake (found version "5.18.0") 
-- Found KF5WidgetsAddons: /usr/lib/x86_64-linux-gnu/cmake/KF5WidgetsAddons/KF5WidgetsAddonsConfig.cmake (found version "5.18.0") 
-- Found KF5: success (found version "5.18.0") found components:  Auth Config ConfigWidgets CoreAddons I18n KIO XmlGui Sonnet Kross Codecs TextWidgets WidgetsAddons 
-- GNU C compiler detected
-- Subtitle Composer plugin path: /usr/lib/subtitlecomposer
-- Found ICU library: /usr/include/x86_64-linux-gnu
-- ICU Library will be used for charset detection.
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'glib-2.0'
--   Found glib-2.0, version 2.48.2
-- Checking for module 'gobject-2.0'
--   Found gobject-2.0, version 2.48.2
-- Found GOBJECT: /usr/lib/x86_64-linux-gnu/libgobject-2.0.so;/usr/lib/x86_64-linux-gnu/libgmodule-2.0.so;/usr/lib/x86_64-linux-gnu/libgthread-2.0.so;/usr/lib/x86_64-linux-gnu/libglib-2.0.so  
-- Checking for module 'gstreamer-1.0'
--   Found gstreamer-1.0, version 1.8.3
-- Found GStreamer: /usr/lib/x86_64-linux-gnu/libgstreamer-1.0.so
-- Found GStreamer: /usr/lib/x86_64-linux-gnu/libgstreamer-1.0.so
-- Checking for module 'gstreamer-plugins-base-1.0'
--   Found gstreamer-plugins-base-1.0, version 1.8.3
-- Checking for module 'gstreamer-audio-1.0'
--   Found gstreamer-audio-1.0, version 1.8.3
-- Checking for module 'gstreamer-cdda-1.0'
--   No package 'gstreamer-cdda-1.0' found
-- Could not find cdda plugin
-- CDDA library: 
-- CDDA include dir: 
-- Checking for module 'gstreamer-netbuffer-1.0'
--   No package 'gstreamer-netbuffer-1.0' found
-- Could not find netbuffer plugin
-- NETBUFFER library: 
-- NETBUFFER include dir: 
-- Checking for module 'gstreamer-pbutils-1.0'
--   Found gstreamer-pbutils-1.0, version 1.8.3
-- Checking for module 'gstreamer-riff-1.0'
--   Found gstreamer-riff-1.0, version 1.8.3
-- Checking for module 'gstreamer-rtp-1.0'
--   Found gstreamer-rtp-1.0, version 1.8.3
-- Checking for module 'gstreamer-rtsp-1.0'
--   Found gstreamer-rtsp-1.0, version 1.8.3
-- Checking for module 'gstreamer-sdp-1.0'
--   Found gstreamer-sdp-1.0, version 1.8.3
-- Checking for module 'gstreamer-tag-1.0'
--   Found gstreamer-tag-1.0, version 1.8.3
-- Checking for module 'gstreamer-video-1.0'
--   Found gstreamer-video-1.0, version 1.8.3
-- Found GStreamer: /usr/lib/x86_64-linux-gnu/libgstreamer-1.0.so (version 1.8.3)
-- Found LibMPV: /usr/lib/x86_64-linux-gnu/libmpv.so (version 1.20.0)
-- Checking for module 'libxine'
--   Found libxine, version 1.2.6
-- Performing Test XINE_XCB_FOUND
-- Performing Test XINE_XCB_FOUND - Success
-- Found Xine: /usr/lib/libxine.so (version 1.2.6)
-- Found XCB_XCB: /usr/lib/x86_64-linux-gnu/libxcb.so (found version "1.11.1") 
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so (found version "1.11.1") found components:  XCB 
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so (version 1.11.1)
-- Xine Player backend will use XCB instead of LibX11
-- Found POCKETSPHINX: pocketsphinx;sphinxbase;sphinxad  
-- Found PocketSphinx version 5prealpha
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ed/Downloads/subtitlecomposer-0.6.5/build
-e \e[1;32mConfigure completed. Now you can:\e[m
cd build
make
sudo make install
ed@duck606 ~/Downloads/subtitlecomposer-0.6.5 $ cd build
ed@duck606 ~/Downloads/subtitlecomposer-0.6.5/build $ make
Scanning dependencies of target pofiles_1
[  0%] Generating subtitlecomposer.gmo
[  0%] Built target pofiles_1
Scanning dependencies of target pofiles_2
[  1%] Generating subtitlecomposer.gmo
[  1%] Built target pofiles_2
Scanning dependencies of target pofiles_3
[  1%] Generating subtitlecomposer.gmo
[  1%] Built target pofiles_3
Scanning dependencies of target pofiles_4
[  2%] Generating subtitlecomposer.gmo
[  2%] Built target pofiles_4
Scanning dependencies of target pofiles_5
[  2%] Generating subtitlecomposer.gmo
[  2%] Built target pofiles_5
Scanning dependencies of target pofiles_6
[  3%] Generating subtitlecomposer.gmo
[  3%] Built target pofiles_6
Scanning dependencies of target pofiles_7
[  3%] Generating subtitlecomposer.gmo
[  3%] Built target pofiles_7
Scanning dependencies of target pofiles_8
[  4%] Generating subtitlecomposer.gmo
[  4%] Built target pofiles_8
Scanning dependencies of target pofiles_9
[  4%] Generating subtitlecomposer.gmo
[  4%] Built target pofiles_9
Scanning dependencies of target pofiles_10
[  5%] Generating subtitlecomposer.gmo
[  5%] Built target pofiles_10
Scanning dependencies of target pofiles_11
[  5%] Generating subtitlecomposer.gmo
[  5%] Built target pofiles_11
Scanning dependencies of target pofiles_12
[  6%] Generating subtitlecomposer.gmo
[  6%] Built target pofiles_12
Scanning dependencies of target pofiles_13
[  6%] Generating subtitlecomposer.gmo
[  6%] Built target pofiles_13
Scanning dependencies of target pofiles_14
[  7%] Generating subtitlecomposer.gmo
[  7%] Built target pofiles_14
Scanning dependencies of target pofiles_15
[  7%] Generating subtitlecomposer.gmo
[  7%] Built target pofiles_15
Scanning dependencies of target vobsubmplayer_automoc
[  7%] Automatic moc for target vobsubmplayer
[  7%] Built target vobsubmplayer_automoc
Scanning dependencies of target vobsubmplayer
[  7%] Building C object src/formats/vobsub/mplayer/CMakeFiles/vobsubmplayer.dir/mp_msg.c.o
[  8%] Building C object src/formats/vobsub/mplayer/CMakeFiles/vobsubmplayer.dir/spudec.c.o
[  8%] Building C object src/formats/vobsub/mplayer/CMakeFiles/vobsubmplayer.dir/unrar_exec.c.o
[  9%] Building C object src/formats/vobsub/mplayer/CMakeFiles/vobsubmplayer.dir/vobsub.c.o
[  9%] Building CXX object src/formats/vobsub/mplayer/CMakeFiles/vobsubmplayer.dir/vobsubmplayer_automoc.cpp.o
[ 10%] Linking CXX static library libvobsubmplayer.a
[ 10%] Built target vobsubmplayer
Scanning dependencies of target subtitlecomposer_automoc
[ 11%] Automatic moc for target subtitlecomposer
Generating moc_fileloadhelper.cpp
Generating moc_actionmanager.cpp
Generating moc_subtitle.cpp
Generating moc_subtitleiterator.cpp
Generating moc_subtitleline.cpp
Generating moc_textdemux.cpp
Generating moc_vobsubinputinitdialog.cpp
Generating moc_vobsubinputprocessdialog.cpp
Generating moc_kcodecactionext.cpp
Generating moc_krecentfilesactionext.cpp
Generating moc_useraction.cpp
Generating moc_application.cpp
Generating moc_configdialog.cpp
Generating moc_errorsconfigwidget.cpp
Generating moc_generalconfigwidget.cpp
Generating moc_playerconfigwidget.cpp
Generating moc_waveformconfigwidget.cpp
Generating moc_currentlinewidget.cpp
Generating moc_actiondialog.cpp
Generating moc_actionwitherrortargetsdialog.cpp
Generating moc_actionwithtargetdialog.cpp
Generating moc_changeframeratedialog.cpp
Generating moc_changetextscasedialog.cpp
Generating moc_checkerrorsdialog.cpp
Generating moc_clearerrorsdialog.cpp
Generating moc_durationlimitsdialog.cpp
Generating moc_intinputdialog.cpp
Generating moc_joinsubtitlesdialog.cpp
Generating moc_opensubtitledialog.cpp
Generating moc_progressdialog.cpp
Generating moc_selectablesubtitledialog.cpp
Generating moc_splitsubtitledialog.cpp
Generating moc_subtitlecolordialog.cpp
Generating moc_syncsubtitlesdialog.cpp
Generating moc_textinputdialog.cpp
Generating moc_errorsdialog.cpp
Generating moc_errorswidget.cpp
Generating moc_lineswidget.cpp
Generating moc_mainwindow.cpp
Generating moc_playerwidget.cpp
Generating moc_scripting_list.cpp
Generating moc_scripting_range.cpp
Generating moc_scripting_rangelist.cpp
Generating moc_scripting_rangesmodule.cpp
Generating moc_scripting_sstring.cpp
Generating moc_scripting_stringsmodule.cpp
Generating moc_scripting_subtitle.cpp
Generating moc_scripting_subtitleline.cpp
Generating moc_scripting_subtitlelinemodule.cpp
Generating moc_scripting_subtitlemodule.cpp
Generating moc_scriptsmanager.cpp
Generating moc_errorfinder.cpp
Generating moc_errortracker.cpp
Generating moc_finder.cpp
Generating moc_replacer.cpp
Generating moc_speller.cpp
Generating moc_translator.cpp
Generating moc_speechplugin.cpp
Generating moc_speechprocessor.cpp
Generating moc_streamprocessor.cpp
Generating moc_playerbackend.cpp
Generating moc_videoplayer.cpp
Generating moc_videowidget.cpp
Generating moc_attachablewidget.cpp
Generating moc_layeredwidget.cpp
Generating moc_pointingslider.cpp
Generating moc_simplerichtextedit.cpp
Generating moc_textoverlaywidget.cpp
Generating moc_timeedit.cpp
Generating moc_treeview.cpp
Generating moc_waveformwidget.cpp
[ 11%] Built target subtitlecomposer_automoc
[ 11%] Generating ui_scriptsmanager.h
[ 11%] Generating scconfig.h, scconfig.cpp
[ 13%] Generating scconfig.moc
/home/ed/Downloads/subtitlecomposer-0.6.5/build/src/main/scconfig.h:0: Note: No relevant classes found. No output generated.
[ 14%] Generating ui_vobsubinputinitdialog.h
[ 14%] Generating ui_vobsubinputprocessdialog.h
[ 15%] Generating ui_errorsconfigwidget.h
[ 15%] Generating ui_generalconfigwidget.h
[ 16%] Generating ui_playerconfigwidget.h
[ 16%] Generating ui_waveformconfigwidget.h
[ 16%] Generating ui_textinputdialog.h
[ 17%] Generating ui_intinputdialog.h
Scanning dependencies of target subtitlecomposer
[ 18%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/common/commondefs.cpp.o
[ 18%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/common/fileloadhelper.cpp.o
[ 19%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/common/filesavehelper.cpp.o
[ 19%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/common/filetrasher.cpp.o
[ 20%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/common/languagecode.cpp.o
[ 20%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/actionmanager.cpp.o
[ 21%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/time.cpp.o
[ 21%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/sstring.cpp.o
[ 22%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/subtitle.cpp.o
[ 22%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/subtitleactions.cpp.o
[ 23%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/subtitleiterator.cpp.o
[ 23%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/subtitleline.cpp.o
[ 25%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/core/subtitlelineactions.cpp.o
[ 25%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/formats/formatmanager.cpp.o
[ 26%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/formats/textdemux/textdemux.cpp.o
[ 26%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/formats/vobsub/vobsubinputinitdialog.cpp.o
[ 27%] Building CXX object src/main/CMakeFiles/subtitlecomposer.dir/__/formats/vobsub/vobsubinputprocessdialog.cpp.o
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp: In member function ‘bool SubtitleComposer::VobSubInputProcessDialog::Frame::processPieces()’:
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:2: error: ‘function’ is not a member of ‘std’
  std::function<void(int,int)> cutPiece = [&](int x, int y){
  ^~~
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:28: error: expression list treated as compound expression in functional cast [-fpermissive]
  std::function<void(int,int)> cutPiece = [&](int x, int y){
                            ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:146:16: error: expected primary-expression before ‘void’
  std::function<void(int,int)> cutPiece = [&](int x, int y){
                ^~~~
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:176:18: error: ‘cutPiece’ was not declared in this scope
     cutPiece(x, y);
                  ^
/home/ed/Downloads/subtitlecomposer-0.6.5/src/formats/vobsub/vobsubinputprocessdialog.cpp:138:6: warning: unused variable ‘bgColor’ [-Wunused-variable]
  int bgColor = pieceBitmap.pixel(0, 0);
      ^~~~~~~
src/main/CMakeFiles/subtitlecomposer.dir/build.make:494: recipe for target 'src/main/CMakeFiles/subtitlecomposer.dir/__/formats/vobsub/vobsubinputprocessdialog.cpp.o' failed
make[2]: *** [src/main/CMakeFiles/subtitlecomposer.dir/__/formats/vobsub/vobsubinputprocessdialog.cpp.o] Error 1
CMakeFiles/Makefile2:1871: recipe for target 'src/main/CMakeFiles/subtitlecomposer.dir/all' failed
make[1]: *** [src/main/CMakeFiles/subtitlecomposer.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by maxrd2** [comment]
Please try pulling newest changes from. Compilation should be fixed with commit https://github.com/maxrd2/subtitlecomposer/commit/6da8b24f922624087d2ac8c2af62b6b4157883d8
Please note that ffmpeg v3 is required now to compile. I found mint binary version on ppa:jonathonf/ffmpeg-3.
Also have made some updates to build instructions on wiki.

**Comment by duck7000** [comment]
First of all a big thank you for your work!

I followed your instructions for dependencies in ubuntu/mint, turns out that i was missing some.
And installed the updated ffmpeg version.

Configure works fine with no errors, but make throws up this error:

```
/home/ed/Downloads/subtitlecomposer-master/src/scripting/scriptsmanager.cpp: In member function ‘void SubtitleComposer::ScriptsManager::editScript(const QString&)’:
/home/ed/Downloads/subtitlecomposer-master/src/scripting/scriptsmanager.cpp:303:84: error: ‘RunFlags’ is not a member of ‘KRun’
  if(!KRun::runUrl(QUrl(m_scripts[scriptName]), "text/plain", app()->mainWindow(), (KRun::RunFlags)0))
                                                                                    ^~~~
src/CMakeFiles/subtitlecomposer.dir/build.make:2102: recipe for target 'src/CMakeFiles/subtitlecomposer.dir/scripting/scriptsmanager.cpp.o' failed
make[2]: *** [src/CMakeFiles/subtitlecomposer.dir/scripting/scriptsmanager.cpp.o] Error 1
CMakeFiles/Makefile2:1129: recipe for target 'src/CMakeFiles/subtitlecomposer.dir/all' failed
make[1]: *** [src/CMakeFiles/subtitlecomposer.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by duck7000** [comment]
@muzena 
Maybe you can try to build a deb file for ubuntu/mint?
I'am trying to build it but i'm not very experienced in Linux, i get the feeling that you have much more experience?

Just to be sure that we are not hunting ghosts here, maybe caused by my inexperience with Linux

**Comment by muzena** [comment]
@duck7000 
I'm trying!
@maxrd2 
I stuck in subtitlecomposer compilation on pocketsphinx, is it old version? I installed it from official repository.
https://launchpad.net/ubuntu/+source/pocketsphinx/0.8.0+real5prealpha-1ubuntu2

This is subtitlecomposer compilation log error:
 
Scanning dependencies of target pocketsphinxasr
[ 91%] Building CXX object src/speechplugins/pocketsphinx/CMakeFiles/pocketsphinxasr.dir/pocketsphinxplugin.cpp.o
[ 91%] Building CXX object src/videoplayerplugins/mpv/CMakeFiles/mpvplayer.dir/mpvplayer_automoc.cpp.o
[ 91%] Building CXX object src/videoplayerplugins/xine/CMakeFiles/xineplayer.dir/xineplayer_automoc.cpp.o
[ 92%] Building CXX object src/videoplayerplugins/mplayer/CMakeFiles/mplayer.dir/__/__/scconfigdummy.cpp.o
[ 93%] Building CXX object src/videoplayerplugins/gstreamer/CMakeFiles/gstplayer.dir/__/__/videoplayer/videowidget.cpp.o
/home/goran/subtitlecomposer/src/subtitlecomposer/src/speechplugins/pocketsphinx/pocketsphinxplugin.cpp: In member function ‘void SubtitleComposer::PocketSphinxPlugin::processUtterance()’:
/home/goran/subtitlecomposer/src/subtitlecomposer/src/speechplugins/pocketsphinx/pocketsphinxplugin.cpp:99:42: error: too few arguments to function ‘ps_seg_t* ps_seg_iter(ps_decoder_t*, int32*)’
  ps_seg_t *iter = ps_seg_iter(m_psDecoder);
                                          ^
In file included from /home/goran/subtitlecomposer/src/subtitlecomposer/src/speechplugins/pocketsphinx/pocketsphinxplugin.cpp:25:0:
/usr/include/pocketsphinx/pocketsphinx.h:462:11: note: declared here
 ps_seg_t *ps_seg_iter(ps_decoder_t *ps, int32 *out_best_score);
           ^
src/speechplugins/pocketsphinx/CMakeFiles/pocketsphinxasr.dir/build.make:62: recipe for target 'src/speechplugins/pocketsphinx/CMakeFiles/pocketsphinxasr.dir/pocketsphinxplugin.cpp.o' failed
make[2]: *** [src/speechplugins/pocketsphinx/CMakeFiles/pocketsphinxasr.dir/pocketsphinxplugin.cpp.o] Error 1
CMakeFiles/Makefile2:2472: recipe for target 'src/speechplugins/pocketsphinx/CMakeFiles/pocketsphinxasr.dir/all' failed
make[1]: *** [src/speechplugins/pocketsphinx/CMakeFiles/pocketsphinxasr.dir/all] Error 2
make[1]: *** Čekam nedovršene poslove....
[ 93%] Building CXX object src/videoplayerplugins/gstreamer/CMakeFiles/gstplayer.dir/__/__/scconfigdummy.cpp.o
[ 94%] Linking CXX shared module mpvplayer.so
[ 94%] Building CXX object src/videoplayerplugins/mplayer/CMakeFiles/mplayer.dir/mplayer_automoc.cpp.o
[ 95%] Linking CXX shared module xineplayer.so
[ 96%] Building CXX object src/videoplayerplugins/gstreamer/CMakeFiles/gstplayer.dir/gstplayer_automoc.cpp.o
[ 96%] Built target mpvplayer
[ 96%] Built target xineplayer
[ 97%] Linking CXX shared module mplayer.so
[ 97%] Linking CXX shared module gstplayer.so
[ 97%] Built target mplayer
[ 97%] Built target gstplayer
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by duck7000** [comment]
@muzena 
Strange that you don't get the error i got, my error is much earlier in the make process.

I use the pocketsphinx version from official mint repo: Version: 0.8.0+real5prealpha-1ubuntu2

**Comment by maxrd2** [comment]
@muzena pocketsphinx issues is same you had in issue #45 same patch still applies

**Comment by muzena** [comment]
@duck7000 
You must add these two repositories, then upgrade system, and try subtitlecomposer compilation.
https://launchpad.net/~kubuntu-ppa/+archive/ubuntu/backports
https://launchpad.net/~jonathonf/+archive/ubuntu/ffmpeg-3

**Comment by muzena** [comment]
@maxrd2 
I successfully generated deb file, but when i try make PPA then i have this error:
https://launchpadlibrarian.net/343824589/buildlog_ubuntu-xenial-amd64.subtitlecomposer_0.6.5-3~xenial_BUILDING.txt.gz

**Comment by muzena** [comment]
@duck7000 
Here is subtitlecomposer deb file:
Change file extension from .zip to .deb then install
[Uploading subtitlecomposer_0.6.5-2~xenial_amd64.zip…]()

**Comment by duck7000** [comment]
@muzena 
I upgradedd my system through the ppa's
And patched the source according to maxrd2 suggestion.

I have build it successfully now!
Although i do get a error when i use checkinstall to build the actual deb file, it installed and worked (as far as tested)

ps your link to the deb file is invalid and leads to this post

**Comment by muzena** [comment]
@duck7000 
Try now download deb file.

**Comment by duck7000** [comment]
@muzena 
Thanks for the deb file!

Funny thing is that the deb file you made differs from mine:
mine is 795,2 kB
yours is 782,7 kB

I don't no the relevance of this haha

**Comment by muzena** [comment]
@duck7000 
Maybe you didn't install build dependencies: libsphinxbase3, libsphinxbase-dev, libpocketsphinx3, libpocketsphinx-dev and install dependencies: pocketsphinx, pocketsphinx-en-us for speech recognition
It is needed for speech recognition
This is @maxrd2  removed from control file in this commit: https://github.com/maxrd2/subtitlecomposer/commit/15b76473e52426057a05984361f2c06111d9bf17

**Comment by duck7000** [comment]
@muzena 
I checked but the dependencies you mentioned are all installed on my system.

Did you resolve your ppa error?

**Comment by duck7000** [comment]
I have still questions
@muzena 
Subtitle composer from your ppa: subtitlecomposer_0.6.3-8-xenial_amd64.deb (it is gone now)
but when installed the about box says 0.6.4 but the filename says 0.6.3?
So what's correct?

After install of the 0.6.5 version from @muzena it is no longer possible to change the icon style. the install removes qt5ct (i had it installed from another ppa) and it removes the system config printer!.
is it possible to get those two back? Try to reinstall won't work, dependency unmet

**Comment by duck7000** [comment]
After some research i found this:
https://github.com/mati75/qt5ct/issues/9

The qt5ct version fo ubuntu 16.10 installs fine and i can set the icon style again.
It only works in terminal on Linuxmint 18.3:
unset QT_STYLE_OVERRIDE
Qt5ct

So far it survives reboot

**Comment by duck7000** [comment]
If there is not to be a official Ubuntu version of this super program this thread can be closed.
The next LTS version of Ubuntu will have version 0.6.5 in the official repo.

Still i would like to compile the latest version but maybe i have to wait for the new LTS version to even try?

Thanks all of you for the help

**Comment by duck7000** [comment]
In the latest version of muzena the speech plugin seems to be missing?

> Speech Recognition failed: Initialization of speech recognition plugin failed(I18N_ARGUMENT_MISSING)
> Code 1(I18N_ARGUMENT_MISSING): (I18N_ARGUMENT_MISSING)

**Comment by muzena** [comment]
@duck7000 
For me Speech Recognition work in Linux Mint 18.3

**Comment by duck7000** [comment]
@muzena 

Okay thanks it works now!
But i would have thought that if i install your .deb it will install all dependencies?
apparently it does not met all dependencies?

**Comment by muzena** [comment]
@duck7000 
Yes, it seems that some dependency missing in control file.

**Comment by duck7000** [comment]
Oke can that be fixed?

Is there a difference in mandatory dependencies and optional dependencies?
Can both be included?

**Comment by duck7000** [comment]
Well this can be closed as well i'll wait for Ubuntu 18.04/linuxmint 19 it will include 0.6.6 in the official repo
